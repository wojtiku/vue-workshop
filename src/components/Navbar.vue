<template>
    <nav class="navbar">
      <router-link class="navbar--brand" to="/">
        {{ appName }}
      </router-link>
      <ul class="navbar--menu">
        <li class="active"><a href="#">Somewhere</a></li>
        <li class="active"><a href="#">Over</a></li>
        <li class="active"><a href="#">The rainbow</a></li>
      </ul>
    </nav>
</template>

<script>

  export default {
    data() {
      return {
        appName: "Vue.js (work)shop"
      }
    }
  }
</script>

<style lang="scss" scoped>
  .navbar {
    @include box();
    line-height: 30px;
    @include clearfix();

    &--brand {
      font-size: 20px;
      float: left;
      margin-right: $gutter;
      color: $c-black;
      &:hover {
        text-decoration: none;
      }
    }

    &--menu {
      float: left;
      list-style: none;
      padding: 0;
      margin: 0;

      li {
        display: inline-block;
        margin-right: $gutter/2;
      }
    }
  }
</style>
