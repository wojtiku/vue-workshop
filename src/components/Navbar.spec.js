import Navbar from '/src/components/Navbar';
import {mount, RouterLinkStub} from '@vue/test-utils';

describe("Navbar", () => {
    it("matches snapshot", () => {
      const wrapper = mount(Navbar, {
        stubs: {
          RouterLink: RouterLinkStub
        }
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
});