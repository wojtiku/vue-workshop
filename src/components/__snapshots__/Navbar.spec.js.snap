// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<nav class="navbar"><a class="navbar--brand">
    Vue.js (work)shop
  </a>
  <ul class="navbar--menu">
    <li class="active"><a href="#">Somewhere</a></li>
    <li class="active"><a href="#">Over</a></li>
    <li class="active"><a href="#">The rainbow</a></li>
  </ul>
</nav>
`;
